### DISCUSSION TITLE
timescaleDB vs influxDB

### DISCUSSION THREAD

**Discussion by pkkrusty** [arendst/Tasmota#19247] [General]
Anyone using timescaleDB in the community? I've been having reliability issues (out of memory errors, slowness, high cpu) as my influxDB instance grows and I am more and more convinced that timescaleDB is the way to go. Slightly less performant with 'small' databases, but based on PostgreSQL, thus much more mature with memory management and error recovery. Also SQL is way easier to deal with than Flux or InfluxQL.

I love the native influxDB driver in Tasmota, but can get similar functionality with tasmota->mqtt->telegraf->timescaledb.

Thoughts from the community? 

Anyone want to write a timescaleDB driver?

**Comment by pkkrusty** [comment]
Some follow up: after more reading, I am currently switching over to VictoriaMetrics. Much smaller RAM footprint so better suited for Raspberry Pi hosts. Can parse InfluxDB line protocol, so essentially a drop in replacement for InfluxDB as far as Tasmota is concerned. I simply changed IfxPort to 8428 and it works. Telegraf can use an InfluxDB output as well, just change the port. 

Only downside is multiple retention policies are not straightforward/available, so I have to default everything to 10 years and eventually figure out how to aggregate my data or just accept a large database.

**Reply by barbudor** [reply]
Interretsing
Thanks for sharing

**Reply by barbudor** [reply]
Is there any migration tools available If I want to move from Influx to VM ?
I have 2 years of data stored

I don't run on a Rpi but x86 N4100 8GB/64GB
I find those small PC much more hasslefree and kind-of cheaper to run than RPi at the end

**Reply by pkkrusty** [reply]
I saw some info on the VM website about transitioning Influx data to VM, so I think it's possible. 

My biggest issue with InfluxDB, aside from large RAM footprint, is that it has a known bug with 32-bit systems where it can't load databases larger than 2.6gb without crashing with out-of-memory error. (That they won't fix) So Raspberry Pi is doomed for a lot of people. Of course can switch to 64-bit kernel, but that means a complete re-build. 

I agree about cheap x86 micro PCs. Cheaper than Raspi and faster. The new ones are even pretty comparable power-wise. I would definitely go that route if I was starting from scratch today.

**Reply by pkkrusty** [reply]
Here's a doc from VM on moving InfluxDB data: Not super straight-forward, but doable. 

https://docs.victoriametrics.com/vmctl.html#migrating-data-from-influxdb-1x

**Comment by stefanbode** [comment]
We are using influxdb for professional usage and do not have any issues. For sure 64Bit is a must, but that should nowadays not a problem. Memory increase only happens at the beginning. Then it is stable. Are you sure your rentention time is ok and you do not store high frequency data forever. The compacting of data is really important on the long run. After a year or two neither ram nor storage should change mentionable. CPU has never been seen as a bottleneck. Only ingestion creates noticeable loads. But a home automation should not able to kick this

**Comment by pkkrusty** [comment]
Agree that with correct retention policies it should not be a problem, but in my case, my preferred retention policies (4 weeks for granular data, 1 year for general data, and 5 years for environmental data) was giving me a DB that was too big for 32-bit raspberry pi with 4gb RAM.  I hadn't yet reached stable state.

Even big orgs get hit with this, i.e. Victron Energy back in May, due to the nature of how Influx deals with queries in memory.

I will see how it goes with VM, but I'm guessing that at least my RAM footprint will be more suited to the Raspberry Pi.
